HappyApp.py

```diff
@@ -34,39 +41,61 @@ def __roll(self):
         
         if len(sadir_names) % 2 != 0:
             self.sadir_left_over = sadir_names[-1]
-            sadir_pairs += sadir_names[-1]
+            sadir_pairs.append(sadir_names[-1])
 
         all_names = sadir_pairs + keva_names
 
         shuffle(all_names)
-        self.queue = Queue(all_names)
+        self.queue = MyQueue(all_names)
 
     def __popQueue(self):
-        self.queue.dequeue()
         if self.queue.isEmpty():
             self.__roll()
-            alert(self.mail,
+            Msg.alert(self.mail,
                   data=self.queue.getQueue(),
                   msg_type=MessageType.new_period)
         else:
-            alert(self.mail,
+            self.queue.dequeue()
+            Msg.alert(self.mail,
                   data=self.queue.head(), 
                   msg_type=MessageType.reminder)
 
-    def run(self, debug):
+    def __debugTime(self, time_period = 1):
```

Comment:
you can add '->' in the signature of a function that returns a value to signal the type of value being returned, making it easier to understand.